shim/src/hashtable.c

```diff
@@ -81,19 +80,18 @@ uint64_t hashtable_get(struct hashtable *h, int key)
     int skip = 0;
     int hash = 0;
     const int length = (1 << h->length_log2);
-    const int mask = (length - 1);
 
     assert(h != NULL);
     assert(key != HASHTABLE_NULL);
 
-    hash = key & mask;
+    hash = key % length;
 
     do
     {
         if (h->keys[hash] == key)
             return (h->values[hash]);
 
-        hash = (hash + 1) & mask;
+        hash = hash + 1;
```

Comment:
We should keep modulo operation here, otherwise we may end up writing outside the bounds of the array.